packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -330,19 +334,22 @@ class IOSDevice extends Device {
       printTrace("Auto selected local port to $localPort");
     }
 
-    int forwardResult = await portForwarder.forward(remotePort,
-        hostPort: localPort).timeout(stepTimeout, onTimeout: () {
-      printTrace("Timeout while atempting to foward port for $serviceName");
-      return null;
-    });
+    int forwardResult = await portForwarder
+      .forward(remoteUri.port, hostPort: localPort)
+      .timeout(stepTimeout, onTimeout: () {
+        printTrace("Timeout while atempting to foward port for $serviceName");
+        return null;
+      });
 
     if (forwardResult == null) {
-      printTrace("Could not foward remote $serviceName port $remotePort to local port $localPort");
+      printTrace("Could not foward remote $serviceName port $remoteUri to local port $localPort");
```

Comment:
each device has it's own way of printing these error messages. If this code was made device-agnostic it will be easier to provide a consistent experience.